### PR TITLE
added rule for failed url request

### PR DESCRIPTION
**Pull request by eriktks**
Solves issue #434 by dealing with failed url requests by testing for fail and in case of success waiting for 60 seconds before repeating the request

### CODE REVIEW

File: harvesting/util.py [hunk 1]
```diff
@@ -67,3 +67,10 @@ def db_connect():
                                connectTimeoutMS=100,
                                serverSelectionTimeoutMS=100
                                )[os.environ.get('DATABASE_NAME')]
+
+def rate_limit_reached(requests_response):
+    try:
+        rate_limit_check = requests_response.headers.get('x-ratelimit-remaining')
+        if int(rate_limit_check) < 10: return True
```
Comment (sverhoeven) [inline harvesting/util.py:74]: As error reported at #434, sometimes the header does not exist and the int conversion fails.

Please add a None check.
```suggestion
        if rate_limit_check is not None and int(rate_limit_check) < 10: return True
```

**Review by sverhoeven — CHANGES_REQUESTED** [review]
Nice refactoring the rate limit check into a function.

Sadly it does not fix my problem of #434, still get the TypeError, see inline suggestion for potential fix. Zenodo is very slow for me at the moment so I could not test fix.

BTW you should have push permissions to this repo so you dont have to use a fork.

File: harvesting/util.py (hunk 1 above)
Comment (eriktks) [reply harvesting/util.py:74]: Thanks for the tips! A new issue #470 was created from a branch which supersedes this pull request. The reported error after the fix could not be reproduced so an extra test was added to verify that the fix worked (harvesting/util_unittest.py)

**Comment by jspaaks** [comment]
Closing this PR since @eriktks says it is superseded by #470. Feel free to reopen if necessary.

**Pull request closed without merging**